### ISSUE TITLE
File Descriptors are left open as deleted

### ISSUE THREAD

**Issue by AvnerCohen** [benoitc/gunicorn#2061]
So, surely this is an ongoing issue for years, happy to provide any additional info.

I guess this is related to:

**Comment by tilgovi** [comment]
@AvnerCohen thank you for reporting this and for taking the time to link to other issues.

Is 28994 the PID there of the arbiter?

**Comment by AvnerCohen** [comment]
Here is another dump  of this:
```sh
$ lsof -c gunicorn | egrep "deleted" | awk {'print $2 '}  | sort | uniq -c | sort -n
     16 29766
     16 29767
     16 29768
     16 29771
     16 29773
     16 29774
     16 29776
     16 29777
     16 29778
     24 29761
```

This is after a restart (using USR2>WINCH>QUIT).
PID = 29761 is the `master` all others are the workers attached to it.
So it looks like all elements are leaking.
What I did notice is that on each "restart" increases the leaked FDs, so maybe it's something around that.

lmk if anything else I can test or provide.
Thanks for the quick response!

**Comment by alochym01** [comment]
i got same error, is there any patches to fix

thank you

**Comment by alochym01** [comment]
@AvnerCohen do you have a work around for this issue

**Comment by AvnerCohen** [comment]
No. No real workaround, just better monitoring to fully stop/strt every now and then.

**Comment by benoitc** [comment]
what happen if you'using latest 19.x or 20.x (master) ?

**Comment by benoitc** [comment]
bump

**Comment by AvnerCohen** [comment]
@benoitc As you can see in the first post,  we already use 19.x (19.8.0).
However, we have recently upgraded to python 3.x and this does no longer seems to be an issue.

So I am guessing , since gunicorn is dropping python  2.x support, this basically can be closed.

Thanks for checking and following up !

**Comment by shantha-Ekanayake** [comment]
Even with the python 3.x I am still facing issues of having temp files in lsof and holding resources. Did anyone find any solution to this problem?

**Comment by javabrett** [comment]
@shantha-Ekanayake this is an old issue that should probably have been closed - can you provide your versions for everything, esp. Python and Gunicorn, and some recorded observations and logs?

**Comment by benoitc** [comment]
closiing in favour of #3250
